src/runtime/gc_custom.go

```diff
@@ -0,0 +1,49 @@
+//go:build gc.custom
+// +build gc.custom
+
+package runtime
+
+// This GC strategy allows an external GC to be plugged in instead of the builtin
+// implementations.
+//
+// The custom implementation must provide the following functions in the runtime package
+// using the go:linkname directive:
+//
+// - func initHeap()
+// - func alloc(size uintptr, layout unsafe.Pointer) unsafe.Pointer
+// - func free(ptr unsafe.Pointer)
+// - func GC()
+// - func KeepAlive(x interface{})
+// - func SetFinalizer(obj interface{}, finalizer interface{})
+//
+// In addition, if targeting wasi, the following functions should be exported for interoperability
+// with wasi libraries that use them. Note, this requires the export directive, not go:linkname.
+//
+// - func malloc(size uintptr) unsafe.Pointer
+// - func free(ptr unsafe.Pointer)
+// - func calloc(nmemb, size uintptr) unsafe.Pointer
+// - func realloc(oldPtr unsafe.Pointer, size uintptr) unsafe.Pointer
```

Comment:
These are normally defined by the runtime, and call `runtime.alloc`/`runtime.free` as needed. Would it be possible to keep it that way and let the custom GC only provide the functions above?

Reply:
We added the custommalloc build tag and this intends to replace it with just the concept of a custom GC.

Replacing malloc with an implementation that doesn't delegate to Go GC (in my case mimalloc) had a dramatic improvement in performance and is important for allowing high performance. I'd be fine with keeping the current tag though if it seems better.